flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/EventUtils.java

```diff
@@ -197,4 +198,19 @@ private static Event buildEvent(
                 .endMetadata()
                 .build();
     }
+
+    private static boolean intervalCheck(Event existing, @Nullable Duration interval) {
+        return interval != null
+                && Instant.now()
+                        .isBefore(
+                                Instant.parse(existing.getLastTimestamp())
+                                        .plusMillis(interval.toMillis()));
+    }
+
+    private static boolean labelCheck(
+            Event existing, Predicate<Map<String, String>> dedupePredicate) {
+        return dedupePredicate == null
+                || (existing.getMetadata() != null
+                        && dedupePredicate.test(existing.getMetadata().getLabels()));
+    }
```

Comment:
I may misunderstand something but seems like labels are basically ignored when the `interval == null` . In that case intervalCheck is always false. Is this intentional?

Reply:
when Interval == null, we don't dedupe. This is intentional. It is commented in the interface method param and unit  tested.
but when the config is null, we use the default value of 30 min. It is also documented and unit tested.

Reply:
makes sense